app/assets/javascripts/member-facing/backbone/fundraiser.js

```diff
@@ -275,32 +378,50 @@ const Fundraiser = Backbone.View.extend(_.extend(
     });
   },
 
-  serializeUserForm () {
-    let list = this.$('form.action-form').serializeArray();
-    let serialized = {}
-    $.each(list, function(ii, field){
-      serialized[field.name] = field.value;
-    });
-    return serialized;
+  serializeUserForm() {
+    const list = this.$('form.action-form').serializeArray();
+    return mapValues(keyBy(list, 'name'), 'value');
   },
 
   readRecurring() {
-    return this.$('input.fundraiser-bar__recurring').prop('checked') ? true : false
+    if(this.oneClickShowing())
+      return !!this.$('input.fundraiser-bar__recurring-one-click').prop('checked');
+    else
+      return !!this.$('input.fundraiser-bar__recurring').prop('checked');
+  },
+
+  readRecurringOneClick() {
+    return !!this.$('input.fundraiser-bar__recurring-one-click').prop('checked');
   },
 
-  disableButton(e) {
+  readStoreInVault() {
+    return this.$('input.fundraiser-bar__store-in-vault').prop('checked');
```

Comment:
`.prop('checked')`will always return a boolean because _checked_ is a [boolean attribute](https://www.w3.org/TR/html4/intro/sgmltut.html#h-3.3.4.2).